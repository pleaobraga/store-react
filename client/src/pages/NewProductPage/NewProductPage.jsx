import React from 'react'
import { useSelector } from 'react-redux'
import { useParams } from 'react-router'

import ErrorPage from '../ErrorPage'
import EditProduct from '../../components/EditProduct'

import './NewProductPage.scss'

const NewProductPage = () => {
  const pageContent = useSelector(state => state.product)
  const { storeName } = useParams()

  const contentPage = () => {
    return (
      <div className="page page-edit-product">
        <h1 className="page-edit-product__title">Create new Product</h1>
        <EditProduct
          currentProduct={{}}
          isEditing={false}
          storeName={storeName}
        />
      </div>
    )
  }

  const { errorContent } = pageContent

  return errorContent ? <ErrorPage /> : contentPage()
}

export default NewProductPage
