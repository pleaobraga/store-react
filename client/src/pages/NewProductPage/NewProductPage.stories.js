import React from 'react'
import { storiesOf } from '@storybook/react'
import NewProductPage from './NewProductPage'

export default storiesOf('Pages | WelcomePage', module).add(
  'default',
  () => <NewProductPage />,
  { info: { inline: true, header: false } }
)
