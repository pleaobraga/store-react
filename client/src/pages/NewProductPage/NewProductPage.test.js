import React from 'react'
import { mount } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'

import {
  mockStore,
  initialStateReducer
} from '../../reducers/__mocks__/reduxMock'
import NewProductPage from './NewProductPage'

const setup = ({ state }) => {
  const newState = {
    product: {
      ...initialStateReducer,
      ...state
    }
  }

  const store = mockStore(newState)

  const contentPage = mount(
    <Provider store={store}>
      <Router>
        <NewProductPage />
      </Router>
    </Provider>
  )

  return contentPage
}

describe('NewProductPage', () => {
  it('should render content page', () => {
    const editProductPage = setup({})
    expect(editProductPage).toMatchSnapshot()
  })

  it('should render error page', () => {
    const editProductPage = setup({ errorContent: true })
    expect(editProductPage).toMatchSnapshot()
  })
})
