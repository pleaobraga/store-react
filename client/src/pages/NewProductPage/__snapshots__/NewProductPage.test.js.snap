// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProductPage should render content page 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <NewProductPage>
        <div
          className="page page-edit-product"
        >
          <h1
            className="page-edit-product__title"
          >
            Create new Product
          </h1>
          <EditProduct
            currentProduct={Object {}}
            isEditing={false}
          >
            <form
              className="edit-product"
            >
              <div
                className="edit-product__message"
              />
              <div
                className="edit-product__values"
              >
                <div
                  className="input-group"
                >
                  <label>
                    Name
                  </label>
                  <input
                    className="name"
                    name="name"
                    onChange={[Function]}
                    placeholder="Type the product name"
                    required={true}
                  />
                </div>
                <div
                  className="input-group"
                >
                  <label>
                    Quantity
                  </label>
                  <input
                    className="quantity"
                    name="quantity"
                    onChange={[Function]}
                    placeholder="qty"
                    required={true}
                    type="number"
                  />
                </div>
                <div
                  className="input-group"
                >
                  <label>
                    Currency
                  </label>
                  <input
                    className="currency"
                    name="currency"
                    onChange={[Function]}
                    placeholder="$"
                    required={true}
                  />
                </div>
                <div
                  className="input-group"
                >
                  <label>
                    Price
                  </label>
                  <input
                    className="price"
                    name="price"
                    onChange={[Function]}
                    required={true}
                  />
                </div>
              </div>
              <div
                className="edit-product__actions"
              >
                <button
                  className="save"
                  onClick={[Function]}
                  type="submit"
                >
                  Save
                </button>
                <button
                  className="cancel"
                  onClick={[Function]}
                >
                  Cancel
                </button>
              </div>
            </form>
          </EditProduct>
        </div>
      </NewProductPage>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`NewProductPage should render error page 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <NewProductPage>
        <div
          className="page page-edit-product"
        >
          <h1
            className="page-edit-product__title"
          >
            Create new Product
          </h1>
          <EditProduct
            currentProduct={Object {}}
            isEditing={false}
          >
            <form
              className="edit-product"
            >
              <div
                className="edit-product__message"
              />
              <div
                className="edit-product__values"
              >
                <div
                  className="input-group"
                >
                  <label>
                    Name
                  </label>
                  <input
                    className="name"
                    name="name"
                    onChange={[Function]}
                    placeholder="Type the product name"
                    required={true}
                  />
                </div>
                <div
                  className="input-group"
                >
                  <label>
                    Quantity
                  </label>
                  <input
                    className="quantity"
                    name="quantity"
                    onChange={[Function]}
                    placeholder="qty"
                    required={true}
                    type="number"
                  />
                </div>
                <div
                  className="input-group"
                >
                  <label>
                    Currency
                  </label>
                  <input
                    className="currency"
                    name="currency"
                    onChange={[Function]}
                    placeholder="$"
                    required={true}
                  />
                </div>
                <div
                  className="input-group"
                >
                  <label>
                    Price
                  </label>
                  <input
                    className="price"
                    name="price"
                    onChange={[Function]}
                    required={true}
                  />
                </div>
              </div>
              <div
                className="edit-product__actions"
              >
                <button
                  className="save"
                  onClick={[Function]}
                  type="submit"
                >
                  Save
                </button>
                <button
                  className="cancel"
                  onClick={[Function]}
                >
                  Cancel
                </button>
              </div>
            </form>
          </EditProduct>
        </div>
      </NewProductPage>
    </Router>
  </BrowserRouter>
</Provider>
`;
